src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -114,6 +130,72 @@ public Localizable getActionBlurb() {
         return SupportPlugin.getComponents();
     }
 
+    public class Bundle {
+        File location;
+        public Bundle(File location) {
+            this.location = location;
+        }
+
+        public File getFile() {
+            return location;
+        }
+
+        public String getName() {
+            return location.getName();
+        }
+
+        public String getHref() {
+            return "bundles/bundle/" + getName();
+        }
+
+        public long getLength() {
+            return location.length();
+        }
+
+        private void doDownload(StaplerRequest req, StaplerResponse rsp) throws IOException {
+            OutputStream os = rsp.getOutputStream();
+            try {
+                rsp.setContentType("application/zip");
+                Util.copyStreamAndClose(new FileInputStream(location), os);
+            } finally {
+                IOUtils.closeQuietly(os);
+            }
+        }
+    }
+
+    public final class BundleList extends ArrayList<Bundle> {
+        public void doBundle(StaplerRequest req, StaplerResponse rsp) throws IOException {
+            String restOfPath = req.getRestOfPath();
+            if(restOfPath.replace('\\', '/').contains("/../")) {
+                // don't serve anything other than files in the artifacts dir
+                rsp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+            if (restOfPath.charAt(0) == '/') {
+                restOfPath = restOfPath.substring(1);
+            }
+            for (Bundle b : this) {
+                if (b.getName().equals(restOfPath)) {
+                    b.doDownload(req, rsp);
+                }
+            }
+        }
+    }
+
+    public @Nonnull BundleList getBundles() {
+        BundleList bundles = new BundleList();
+        for (File f : SupportPlugin.getRootDirectory().listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.contains(".zip");
```

Comment:
so `foo.zip.blah` should be accepted?  `endsWith` is probably more correct and more performant.